src/main/java/org/sqlite/core/NativeDB.java

```diff
@@ -559,4 +559,24 @@ long getUpdateListener() {
     long getProgressHandler() {
         return progressHandler;
     }
+
+    @Override
+    public native synchronized long serializeSize(String schema);
+
+    @Override
+    public synchronized ByteBuffer serialize(String schema) {
+        long size = serializeSize(schema);
+        if (size > Integer.MAX_VALUE) {
```

Comment:
why is that a problem?

Reply:
ByteBuffer is limited to max int

Reply:
can we use something else in that case ? that seems very small

Reply:
It's all meant for in memory databases so not that small.
We could use an array of buffers. That would not be a very elegant API though.
I could wrap it with an InputStream so the end user will access it with a more standard approach. Could support both APIs. buffer for small databases and InputStream for larger?
wdyt?

Reply:
> It's all meant for in memory databases so not that small.

Not sure to understand this statement. This can be used to serialize any database, not just in-memory, no? Or did i misunderstood something?

>  We could use an array of buffers. That would not be a very elegant API though.

do we have to use a Buffer ? Why not just a byte array?

Reply:
Theoretically you could Serialize a file into memory but that would be very inefficient. Makes much more sense to just read the file. Deserialize is by definition memory only.
The max sqlite default for this mode is only 1,073,741,824 (1gb).
I am checking if I can increase it.
There are good reasons to switch to byte[] (might be faster oveall, see comment below). However, it is also limited to max int size.

Reply:
More info here: I've managed to raise the limit but anything beyond 2gb results in a malloc error from sqlite.
Turns out it does want to allocate more than 2gb of continuous space. I will ask the sqlite folks if there is a way around it but for now I think we should leave it at that.

Reply:
Asked and answered by sqlite maintainer. 2gb is a hard coded limit in the library, no way to change it, not even with a compilation flag.

Reply:
might be worth to leave a comment in the code about it

Reply:
Done. Also added the limit to the javadoc comment.

Reply:
I believe this limit is slightly wrong.
See: https://stackoverflow.com/questions/3038392/do-java-arrays-have-a-maximum-size
See: https://github.com/OpenHFT/Chronicle-Bytes/blob/a02a79ed8ce9900f8748ad6f3ce801fa241844d1/src/main/java/net/openhft/chronicle/bytes/Bytes.java#L110

I would go with the more conservative limit that openhft suggests. The real limit of a byte array size is JVM dependent. In addition, I don't think this is reachable anyways due to the sqlite malloc limit.

Reply:
@pyckle I think every module should take care of its own limitations. It is true that the limitation on the sqlite is a bit more lower than the java side so this shouldn't happen (added a comment to that effect).

I did develop a unit test that shows the database can grow up to ~2gb. It of course needs at least 4gb of free memory and takes around 30 seconds. Do you think I should commit it? I thought it was a bit extreme and could fail on some test environments.

Reply:
> I did develop a unit test that shows the database can grow up to ~2gb. It of course needs at least 4gb of free memory and takes around 30 seconds. Do you think I should commit it? I thought it was a bit extreme and could fail on some test environments.

the github workers only have 7gb of ram and 15gb of disk space (total, not available)

Reply:
This is up to @gotson as he's the maintainer, but I would suggest committing it with an `@Ignore` annotation. That way we won't have stability issues on CI and if issues with large DB size arise in the future, one can run this test locally.

Reply:
Done as you suggested, testVeryLarge is committed and Disabled